backend/src/utils/pokemon-utils/pokemon-utils.ts

```diff
@@ -1,10 +1,11 @@
+import { ProgrammingError } from '../../domain/error/programming/programming-error';
 import { OPTIMAL_POKEDEX, Pokemon } from '../../domain/pokemon/pokemon';
 import { getBerriesForFilter } from '../berry-utils/berry-utils';
 
 export function getPokemonForName(name: string): Pokemon {
   const pokemon = OPTIMAL_POKEDEX.find((pokemon) => pokemon.name === name);
   if (!pokemon) {
-    throw new Error(`Programmer error: Can't find Pokemon coming from DB with name ${name}`);
+    throw new ProgrammingError(`Programmer error: Can't find Pokemon coming from DB with name ${name}`);
```

Comment:
In this specific case we don't have to mention twice that it's a programmer error, so we can just throw:
```typescript
throw new ProgrammingError(`Can't find Pokemon coming from DB with name ${name}`);
```